geomstats/information_geometry/binomial.py

```diff
@@ -0,0 +1,196 @@
+        atol : float
+            Tolerance to evaluate positivity.
+
+        Returns
+        -------
+        projected : array-like, shape=[...,]
+            Projected point.
+        """
+        point = gs.array(point, gs.float32)
+        projected = gs.where(
+            gs.logical_or(point < atol, point > 1 - atol),
+            (1 - atol) * gs.cast((point > 1 - atol), gs.float32)
+            + atol * gs.cast((point < atol), gs.float32),
+            point,
+        )
+        return gs.squeeze(projected)
+
+    def sample(self, point, n_samples=1):
+        """Sample from the binomial distribution.
+
+        Sample from the binomial distribution with parameter provided by point.
+
+        Parameters
+        ----------
+        point : array-like, shape=[...]
+            Point representing a binomial distribution.
+        n_samples : int
+            Number of points to sample with each pair of parameters in point.
+            Optional, default: 1.
+
+        Returns
+        -------
+        samples : array-like, shape=[...]
+            Sample from binomial distributions.
+        """
+        geomstats.errors.check_belongs(point, self)
+        point = gs.to_ndarray(point, to_ndim=1)
+        samples = gs.array([binom.rvs(self.n_draws, point) for i in range(n_samples)])
+        return gs.squeeze(gs.transpose(samples))
+
+    def point_to_pmf(self, point):
+        """Compute pmf associated to point.
+
+        Compute the probability density function of the binomial
+        distribution with parameters provided by point.
+
+        Parameters
+        ----------
+        point : array-like, shape=[...,]
+            Point representing a binomial distribution (probability of success).
+
+        Returns
+        -------
+        pmf : function
+            Probability density function of the binomial distribution with
+            parameters provided by point.
+        """
+        geomstats.errors.check_belongs(point, self)
+
+        def pmf(k):
+            """Generate parameterized function for binomial pmf.
+
+            Parameters
+            ----------
+            k : array-like, shape=[n_points,]
+                Integers in {0, ..., n_draws} at which to
+                compute the probability mass function.
+            """
+            k = gs.array(k, gs.float32)
+            k = gs.to_ndarray(k, to_ndim=1)
+
+            pmf_at_k = gs.array(
+                [
+                    gs.array(binom.pmf(k, self.n_draws, param))
+                    for param in gs.to_ndarray(point, 1)
+                ]
+            )
+
+            return gs.squeeze(pmf_at_k)
+
+        return pmf
+
+
+class BinomialFisherRaoMetric(RiemannianMetric):
+    """Class for the Fisher information metric on binomial distributions.
+
+    References
+    ----------
+    Atkinson, C., & Mitchell, A. F. (1981). Rao's distance measure.
+    Sankhyā: The Indian Journal of Statistics, Series A, 345-365.
+    """
```

Comment:
.. [AM1981] Atkinson, C., & Mitchell, A. F. (1981). Rao's distance measure.
    Sankhyā: The Indian Journal of Statistics, Series A, 345-365.